addons/calendar/models/calendar.py

```diff
@@ -1258,20 +1281,100 @@ def detach_recurring_event(self, values=None):
+        }
+
+    @api.multi
+    def get_split_recurring_event(self):
+        """ Duplicate a calendar.event if we want to update all the virtual events from a fixed date.
+            In case of end type with count:
+                It will split the count between past events and remaining events. It will assign pas event count to original event
+                and the remaining count to the new event
+            If end type is a final_date:
+                It will set the final date of the original meeting to the selected event start date.
+                It will set the final date for the new event to the original final date.
+            Special case: If we selected the first event of the recurrence we will return the real id because there is no need to split between future and past events.
+            (update future events from the first event = update all events)
+            :return: A new record set containing the recurrence for future events.
+        """
+        real_id, meeting_origin, data = self.get_origin_meeting_and_schedule_fields()
+
+        reccurent_dates = meeting_origin._get_recurrent_date_by_event()
+        # If we want all the future event from the first event, that means we want all events and there is no need to split it.
+        real_start = pytz.UTC.localize(fields.Datetime.from_string(data['start'])).astimezone(pytz.UTC)
+        try:
+            event_position = reccurent_dates.index(real_start)
+        except:
+            raise UserError('Save your recurrent settingq before choosing an action')
+        if event_position == 0:
+            return self.browse(real_id)
+
+        if data.get('rrule'):
+            # if use final date, split old reccurent event until self.start
+            if meeting_origin.end_type == 'end_date':
+                # The final date of the old event is equal to the start date of the new event.
+                data.update['final_date'] = meeting_origin.final_date
+                meeting_origin.final_date = data['start']
+            elif meeting_origin.end_type == 'count':
+                # If we use count, we should try to cut the count in two part.
+                old_count, new_count = self._compute_count_division(meeting_origin, reccurent_dates)
+                meeting_origin.count = old_count
+                data['count'] = new_count
+            data.update(
+                recurrent_id=real_id,
+                recurrent_id_date=data.get('start'),
+                rrule='',
+            )
+
+            # do not copy the id
+            data.pop('id')
+        return meeting_origin.copy(default=data)
+
+    @api.multi
+    def action_future_recurring_event(self):
+        """ Action used in the view to update all the future events from a selected date(a virtual event).
+            it will call the get_split_recurring_event method that will return a calendar event record that
+            has a recurrent settings for all future events.
+        """
```

Comment:
Used to update the next occurrences of a recurrent event according to one of its events. It'll split the two sets of occurrences in two real calendar events and open the form view of the second one.